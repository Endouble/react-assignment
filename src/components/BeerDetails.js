import React from 'react';
import PropTypes from 'prop-types';
import { Header, Image, Modal } from 'semantic-ui-react';

function BeerDetails(props) {
    const { beer, onCloseDetail } = props;

    return (
        <Modal open closeIcon onClose={onCloseDetail}>
            <Modal.Header>{beer.name}</Modal.Header>
            <Modal.Content image scrolling>
                <Image wrapped size='medium' src={beer.image_url} alt={beer.name} />
                <Modal.Description>
                    <Header>{beer.tagline}</Header>
                    <Header as='h4'>Description</Header>
                    <p>{beer.description}</p>
                    <Header as='h4'>Food Pairing</Header>
                    <p>{beer.food_pairing}</p>
                    <Header as='h4'>Brewers Tips</Header>
                    <p>{beer.brewers_tips}</p>
                </Modal.Description>
            </Modal.Content>
        </Modal>
    );
}

BeerDetails.propTypes = {
    beer: PropTypes.shape({
        name: PropTypes.string.isRequired,
        tagline: PropTypes.string.isRequired,
        description: PropTypes.string.isRequired,
        image_url: PropTypes.string.isRequired,
        food_pairing: PropTypes.string.isRequired,
        brewers_tips: PropTypes.string.isRequired,
    }).isRequired,
    onCloseDetail: PropTypes.func.isRequired,
};

export default BeerDetails;
