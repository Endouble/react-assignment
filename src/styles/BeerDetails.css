@media all and (-ms-high-contrast: none), (-ms-high-contrast: active) {
    .ui .modal{
        top: 10%;
        left: 2%;
    }
 }

/* Medium devices (tablets, 768px and up) */
@media (min-width: 768px) and (-ms-high-contrast: none), (-ms-high-contrast: active) { 
    .ui .modal{
        top: 10%;
        left: 5%;
    }
}

/* Large devices (desktops, 992px and up) */
@media (min-width: 992px) and (-ms-high-contrast: none), (-ms-high-contrast: active) {
    .ui .modal{
        top:10%;
        left:10%;
    }
}

/* Extra large devices (large desktops, 1200px and up) */
@media (min-width: 1200px) and (-ms-high-contrast: none), (-ms-high-contrast: active) {
    .ui .modal{
        top:10%;
        left:25%;
    }
}