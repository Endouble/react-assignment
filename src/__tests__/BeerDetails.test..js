import React from 'react';
import { shallow } from 'enzyme';
import BeerDetails from '../components/BeerDetails';

describe('BeerDetails Unit Test', () => {
    let props;
    let shallowedBeerDetails;

    const getBeerDetails = () => {
        if (!shallowedBeerDetails) {
            shallowedBeerDetails = shallow(<BeerDetails {...props} />);
        }
        return shallowedBeerDetails;
    };

    beforeEach(() => {
        props = {
            beer: {
                name: 'Test Beer',
                tagline: 'A beer for testing',
                description: 'Just a mock beer for testing',
                image_url: 'https://testurl.com/beer/image.png',
                food_pairing: 'Every snack you can imagine',
                brewers_tips: 'Drink carefully',
            },
            onCloseDetail: jest.fn()
        };
    });

    test('Component should render without crashing', () => {
        expect(getBeerDetails()).toBeDefined();
    });

    test('Component should render correctly', () => {
        expect(getBeerDetails()).toMatchSnapshot();
    });
});
