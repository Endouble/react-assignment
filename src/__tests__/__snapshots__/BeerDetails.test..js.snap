// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BeerDetails Unit Test Component should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BeerDetails
    beer={
      Object {
        "brewers_tips": "Drink carefully",
        "description": "Just a mock beer for testing",
        "food_pairing": Array [
          "Every snack you can imagine",
        ],
        "image_url": "https://testurl.com/beer/image.png",
        "name": "Test Beer",
        "tagline": "A beer for testing",
      }
    }
    onCloseDetail={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "centered": true,
      "children": Array [
        <ModalHeader>
          Test Beer
        </ModalHeader>,
        <ModalContent
          image={true}
          scrolling={true}
        >
          <Image
            alt="Test Beer"
            as="img"
            size="medium"
            src="https://testurl.com/beer/image.png"
            ui={true}
            wrapped={true}
          />
          <ModalDescription>
            <Header>
              A beer for testing
            </Header>
            <Header
              as="h4"
            >
              Description
            </Header>
            <p>
              Just a mock beer for testing
            </p>
            <Header
              as="h4"
            >
              Food Pairing
            </Header>
            <ul>
              <li>
                Every snack you can imagine
              </li>
            </ul>
            <Header
              as="h4"
            >
              Brewers Tips
            </Header>
            <p>
              Drink carefully
            </p>
          </ModalDescription>
        </ModalContent>,
      ],
      "closeIcon": true,
      "closeOnDimmerClick": true,
      "closeOnDocumentClick": false,
      "dimmer": true,
      "eventPool": "Modal",
      "onClose": [MockFunction],
      "open": true,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Test Beer",
        },
        "ref": null,
        "rendered": "Test Beer",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <Image
              alt="Test Beer"
              as="img"
              size="medium"
              src="https://testurl.com/beer/image.png"
              ui={true}
              wrapped={true}
            />,
            <ModalDescription>
              <Header>
                A beer for testing
              </Header>
              <Header
                as="h4"
              >
                Description
              </Header>
              <p>
                Just a mock beer for testing
              </p>
              <Header
                as="h4"
              >
                Food Pairing
              </Header>
              <ul>
                <li>
                  Every snack you can imagine
                </li>
              </ul>
              <Header
                as="h4"
              >
                Brewers Tips
              </Header>
              <p>
                Drink carefully
              </p>
            </ModalDescription>,
          ],
          "image": true,
          "scrolling": true,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "alt": "Test Beer",
              "as": "img",
              "size": "medium",
              "src": "https://testurl.com/beer/image.png",
              "ui": true,
              "wrapped": true,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <Header>
                  A beer for testing
                </Header>,
                <Header
                  as="h4"
                >
                  Description
                </Header>,
                <p>
                  Just a mock beer for testing
                </p>,
                <Header
                  as="h4"
                >
                  Food Pairing
                </Header>,
                <ul>
                  <li>
                    Every snack you can imagine
                  </li>
                </ul>,
                <Header
                  as="h4"
                >
                  Brewers Tips
                </Header>,
                <p>
                  Drink carefully
                </p>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": "A beer for testing",
                },
                "ref": null,
                "rendered": "A beer for testing",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "as": "h4",
                  "children": "Description",
                },
                "ref": null,
                "rendered": "Description",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Just a mock beer for testing",
                },
                "ref": null,
                "rendered": "Just a mock beer for testing",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "as": "h4",
                  "children": "Food Pairing",
                },
                "ref": null,
                "rendered": "Food Pairing",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <li>
                      Every snack you can imagine
                    </li>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": "0food",
                    "nodeType": "host",
                    "props": Object {
                      "children": "Every snack you can imagine",
                    },
                    "ref": null,
                    "rendered": "Every snack you can imagine",
                    "type": "li",
                  },
                ],
                "type": "ul",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "as": "h4",
                  "children": "Brewers Tips",
                },
                "ref": null,
                "rendered": "Brewers Tips",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Drink carefully",
                },
                "ref": null,
                "rendered": "Drink carefully",
                "type": "p",
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "centered": true,
        "children": Array [
          <ModalHeader>
            Test Beer
          </ModalHeader>,
          <ModalContent
            image={true}
            scrolling={true}
          >
            <Image
              alt="Test Beer"
              as="img"
              size="medium"
              src="https://testurl.com/beer/image.png"
              ui={true}
              wrapped={true}
            />
            <ModalDescription>
              <Header>
                A beer for testing
              </Header>
              <Header
                as="h4"
              >
                Description
              </Header>
              <p>
                Just a mock beer for testing
              </p>
              <Header
                as="h4"
              >
                Food Pairing
              </Header>
              <ul>
                <li>
                  Every snack you can imagine
                </li>
              </ul>
              <Header
                as="h4"
              >
                Brewers Tips
              </Header>
              <p>
                Drink carefully
              </p>
            </ModalDescription>
          </ModalContent>,
        ],
        "closeIcon": true,
        "closeOnDimmerClick": true,
        "closeOnDocumentClick": false,
        "dimmer": true,
        "eventPool": "Modal",
        "onClose": [MockFunction],
        "open": true,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Test Beer",
          },
          "ref": null,
          "rendered": "Test Beer",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <Image
                alt="Test Beer"
                as="img"
                size="medium"
                src="https://testurl.com/beer/image.png"
                ui={true}
                wrapped={true}
              />,
              <ModalDescription>
                <Header>
                  A beer for testing
                </Header>
                <Header
                  as="h4"
                >
                  Description
                </Header>
                <p>
                  Just a mock beer for testing
                </p>
                <Header
                  as="h4"
                >
                  Food Pairing
                </Header>
                <ul>
                  <li>
                    Every snack you can imagine
                  </li>
                </ul>
                <Header
                  as="h4"
                >
                  Brewers Tips
                </Header>
                <p>
                  Drink carefully
                </p>
              </ModalDescription>,
            ],
            "image": true,
            "scrolling": true,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "alt": "Test Beer",
                "as": "img",
                "size": "medium",
                "src": "https://testurl.com/beer/image.png",
                "ui": true,
                "wrapped": true,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <Header>
                    A beer for testing
                  </Header>,
                  <Header
                    as="h4"
                  >
                    Description
                  </Header>,
                  <p>
                    Just a mock beer for testing
                  </p>,
                  <Header
                    as="h4"
                  >
                    Food Pairing
                  </Header>,
                  <ul>
                    <li>
                      Every snack you can imagine
                    </li>
                  </ul>,
                  <Header
                    as="h4"
                  >
                    Brewers Tips
                  </Header>,
                  <p>
                    Drink carefully
                  </p>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": "A beer for testing",
                  },
                  "ref": null,
                  "rendered": "A beer for testing",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "as": "h4",
                    "children": "Description",
                  },
                  "ref": null,
                  "rendered": "Description",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Just a mock beer for testing",
                  },
                  "ref": null,
                  "rendered": "Just a mock beer for testing",
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "as": "h4",
                    "children": "Food Pairing",
                  },
                  "ref": null,
                  "rendered": "Food Pairing",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <li>
                        Every snack you can imagine
                      </li>,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": "0food",
                      "nodeType": "host",
                      "props": Object {
                        "children": "Every snack you can imagine",
                      },
                      "ref": null,
                      "rendered": "Every snack you can imagine",
                      "type": "li",
                    },
                  ],
                  "type": "ul",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "as": "h4",
                    "children": "Brewers Tips",
                  },
                  "ref": null,
                  "rendered": "Brewers Tips",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Drink carefully",
                  },
                  "ref": null,
                  "rendered": "Drink carefully",
                  "type": "p",
                },
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
